pkg/await/apps_deployment.go

```diff
@@ -527,6 +590,26 @@ func (dia *deploymentInitAwaiter) processPodEvent(event watch.Event) {
 	dia.pods[podName] = pod
 }
 
+func (dia *deploymentInitAwaiter) processPersistentVolumeClaimsEvent(event watch.Event) {
+	pvc, isUnstructured := event.Object.(*unstructured.Unstructured)
+	if !isUnstructured {
+		glog.V(3).Infof("PersistentVolumeClaim watch received unknown object type %q",
+			reflect.TypeOf(pvc))
+		return
+	}
+
+	glog.V(3).Infof("Received update for PersistentVolumeClaim %q", pvc.GetName())
+
+	// If Pod was deleted, remove it from our aggregated checkers.
+	uid := string(pvc.GetUID())
```

Comment:
Ha, I had forgotten about this. Historically we've been using `[namespace/]<name>` as the canonical ID, but maybe we should be using this instead everywhere. cc @lblackstone

Reply:
Yeah, UID seems nice. Probably should open an issue to track.

Reply:
Ya, I went with UID to avoid any unforeseen `namespace/name` conflicts